proposals/lip-0053.md

```diff
@@ -133,28 +144,247 @@ The following constants are used throughout the document:
-| `COMMAND_ID_SIDECHAIN_CCU`                               | uint32 | 2                       |
-| `COMMAND_ID_MAINCHAIN_CCU`                               | uint32 | 3                       |
+| `COMMAND_ID_SIDECHAIN_CCU`                               | bytes  | 0x0002                  |
+| `COMMAND_ID_MAINCHAIN_CCU`                               | bytes  | 0x0003                  |
 | **Chain Status**                                         |        |                         |
-| `CHAIN_REGISTERED`                                       | uint32 | 0                       |
-| `CHAIN_ACTIVE`                                           | uint32 | 1                       |  
-| `CHAIN_TERMINATED`                                       | uint32 | 2                       |
+| `CHAIN_STATUS_REGISTERED`                                | uint32 | 0                       |
+| `CHAIN_STATUS_ACTIVE`                                    | uint32 | 1                       |  
+| `CHAIN_STATUS_TERMINATED`                                | uint32 | 2                       |
 | **Message Tags**                                         |        |                         |
 | `MESSAGE_TAG_CERTIFICATE `                               | bytes  | "LSK_CE_" ASCII-encoded |
 
 Several of those constants are shared with the other LIPs defining the Interoperability module and all of the needed constants for the Interoperability module are defined in [LIP 0045][lip-0045]. That LIP should be considered correct if a value stated here differs.
 
-#### uint32be Function
+### Type Definition
+
+| Name               | Type    | Validation            | Description                     |
+|--------------------|---------|-----------------------|---------------------------------|
+| `Address` | bytes   | Must be of length `ADDRESS_LENGTH`. | Address of an account.   |
+| `ModuleID` | bytes   | Must be of length `MODULE_ID_LENGTH`. | ID of a module.   |
+| `CrossChainCommandID` | bytes   | Must be of length `CROSS_CHAIN_COMMAND_ID_LENGTH`. | ID of a cross-chain command.   |
+| `ChainID` | bytes   | Must be of length `CHAIN_ID_LENGTH`. | ID of a chain.   |
+| `CCM` | object   | Must follow the `crossChainMessageSchema` schema. | The type of cross-chain messages.   |
+| `MerkleRoot` | bytes   | Must be of length `MERKLE_ROOT_LENGTH`. | The root of a Merkle tree.   |
+| `PublicKeyEd25519` | bytes | Must be of length `ED25519_PUBLIC_KEY_LENGTH`. | Used for Ed25519 public keys. |
+| `PublicKeyBLS` | bytes | Must be of length `BLS_PUBLIC_KEY_LENGTH`. | Used for BLS keys. |
+| `ActiveValidator` | object | Must have `blsKey: PublicKeyBLS` and `bftWeight: uint64` properties. | The BLS key and BFT weight of an active validator. |
+
+### Auxiliary Functions
+
+The following auxiliary functions are used internally by the Interoperability module to verify and execute cross-chain updates.
+
+#### verifyLivenessCondition
+
+```python
+def verifyLivenessCondition(trs: Transaction) -> None:
```

Comment:
Can we maybe introduce the type `CCU` or something similar and use that instead. It was unclear to me first, why certain properties exist, especially that the schema of a CCU has not been introduced.

Reply:
done